src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -13,62 +13,22 @@ internal static class BestTypeInferrer
         public static NullableAnnotation GetNullableAnnotation(TypeSymbol bestType, ArrayBuilder<TypeSymbolWithAnnotations> types)
         {
             bool bestTypeIsPossiblyNullableReferenceTypeTypeParameter = bestType.IsPossiblyNullableReferenceTypeTypeParameter();
-            NullableAnnotation? result = null;
+            NullableAnnotation result = NullableAnnotation.NotAnnotated;
             foreach (var type in types)
             {
                 if (type.IsNull)
                 {
                     // https://github.com/dotnet/roslyn/issues/27961 Should ignore untyped
                     // expressions such as unbound lambdas and typeless tuples.
-                    result = NullableAnnotation.Nullable;
+                    result = NullableAnnotation.Annotated;
                     continue;
```

Comment:
For an untyped expression such as `null`, Nullable (i.e. possibly null) makes sense.  For an untyped expression such as a lambda, NotAnnotated (i.e. not null) makes sense.  Without that context I don't think you can do anything other than `continue`.

Reply:
In other words, why can't you just remove the entire enclosing `if` statement and its contents?

---
In reply to: [243853209](https://github.com/dotnet/roslyn/pull/31813#discussion_r243853209) [](ancestors = 243853209)

Reply:
I had tried the change you suggested, but got 5 or 6 test regressions that will need further investigation. My suspicion is that we'll need a way to pass a TSWA with an annotation but a null type. Currently that is not allowed by TSWA.

We already have a tracking issue for this, so I'll leave as-is for now.